src/components/core/common-behaviors/link-base-element.ts

```diff
@@ -0,0 +1,114 @@
+
+/** The interface contains attributes that can be set on an <a> tag. */
+export interface LinkProperties {
+  href?: string;
+  target?: LinkTargetType | string;
+  rel?: string;
+  download?: boolean;
+  disabled?: boolean;
+}
+
+function filterUndefined(...objects: Record<string, string | undefined>[]): Record<string, string> {
+  const result: Record<string, string> = {};
+  for (const object of objects) {
+    for (const [key, value] of Object.entries(object)) {
+      if (value !== undefined) {
+        result[key] = value as string;
+      }
+    }
+  }
+  return result;
+}
+
+/**
+ * Lists all attributes for a link; undefined/null properties are not set.
+ * @param linkProperties link properties
+ */
+function getLinkAttributeList(linkProperties: LinkProperties): Record<string, string> {
+  const baseAttributeList = {
+    role: 'presentation',
+    tabIndex: '-1',
+  };
+
+  return !linkProperties.href
+    ? baseAttributeList
+    : filterUndefined(baseAttributeList, {
+        href: linkProperties.href,
+        download: linkProperties.download ? '' : undefined,
+        target: linkProperties.target,
+        rel: linkProperties.rel
+          ? linkProperties.rel
+          : linkProperties.target === '_blank'
+            ? 'external noopener nofollow'
+            : undefined,
+      });
+}
+
+/** A component that implements LinkProperties should use this interface to set useful variables for render function. */
+export interface LinkRenderVariables {
+  /** The tag's name rendered by the component. */
+  tagName: 'a' | 'span';
+
+  /** The tag's attributes. */
+  attributes: Record<string, string>;
+
+  /** The host's attributes. */
+  hostAttributes: Record<string, string | undefined>;
+}
+
+/** Set default render variables for anchor-like elements. */
+function resolveLinkRenderVariables(properties: LinkProperties): LinkRenderVariables {
+  return {
+    tagName: 'a',
+    attributes: getLinkAttributeList(properties),
+    hostAttributes: hostProperties('link', properties.disabled),
+  };
+}
+
+/** Set default render variables for anchor-like elements when the element is static. */
+function resolveStaticRenderVariables(): LinkRenderVariables {
+  return {
+    tagName: 'span',
+    attributes: {},
+    hostAttributes: { dir: getDocumentWritingMode() },
+  };
+}
+
+/** Sets default render variables for anchor-like elements. */
+export function resolveLinkOrStaticRenderVariables(
+  properties: LinkProperties & Partial<IsStaticProperty>,
+): LinkRenderVariables {
+  return properties.isStatic || !properties.href
+    ? resolveStaticRenderVariables()
+    : resolveLinkRenderVariables(properties);
+}
+
+/** Returns true, if href is set and target is _blank. */
+export function targetsNewWindow(properties: LinkProperties): boolean {
+  return !!properties.href && properties.target === '_blank';
+}
+
+/** Link base class. */
+export class SbbLinkBaseElement extends LitElement implements LinkProperties {
```

Comment:
I think we should mark these base classes as abstract.

Reply:
It works if I change all the mixins, replacing `Constructor` with `AbstractConstructor` and marking the inner class as abstract too.
Is it ok for you?
![image](https://github.com/lyne-design-system/lyne-components/assets/101575400/433ce6a5-aca1-44ac-9035-5261df7185f7)

Reply:
That would be okay for me. What do you think?
My motivation is to mark it as abstract, as we do not expect instances of these base classes.

Reply:
It's fine for me, i did it in commit [5693bd8](https://github.com/lyne-design-system/lyne-components/pull/2364/commits/5693bd87b05a516ff9d8b93531c7fef2e0916883)